Applications/spire/source/charting/cached_chart_model.cpp

```diff
@@ -16,24 +16,24 @@ ChartValue::Type CachedChartModel::get_y_axis_type() const {
 
 QtPromise<std::vector<Candlestick>> CachedChartModel::load(ChartValue first,
     ChartValue last) {
-  auto range = ChartRange({first, last});
+  auto range = ChartRange{first, last};
   auto range_iterator = std::lower_bound(m_ranges.begin(), m_ranges.end(),
-    range, [=] (const auto& value, const auto& search_value) {
-      return value.m_start >= search_value.m_start &&
-        value.m_end <= search_value.m_end;
+    range, [] (const auto& value, const auto& search_value) {
+      return value.m_start > search_value.m_start &&
+        value.m_end < search_value.m_end;
     });
   if(range_iterator != m_ranges.end() && (*range_iterator).m_start <= first &&
       (*range_iterator).m_end >= last) {
     return QtPromise([=] {
       auto first_iterator = std::lower_bound(m_loaded_data.begin(),
         m_loaded_data.end(), first,
-        [=] (const auto& value, const auto& search_value) {
+        [] (const auto& value, const auto& search_value) {
           return value.GetStart() < search_value;
       });
-      auto last_iterator = std::lower_bound(m_loaded_data.begin(),
+      auto last_iterator = std::upper_bound(m_loaded_data.begin(),
         m_loaded_data.end(), last,
-        [=] (const auto& value, const auto& search_value) {
-          return value.GetEnd() <= search_value;
+        [] (const auto& search_value, const auto& value) {
+          return value.GetEnd() > search_value;
```

Comment:
Orderings should be defined in terms of ```operator <``` rather than ```operator >```. Reverse the operators to make it that way.

Reply:
So is this valid:

return search_value < value.GetEnd();

I can't find a way to do it using value.GetStart() for the comparison.

Reply:
It's valid in the sense that ```a < b``` is semantically equal to ```b > a```.

Reply:
But it's not valid here?

Reply:
In terms of style guide, yes it's valid but it's likely buggy since you're defining multiple unrelated order relations and I don't see any part of the implementation that guarantees the ordering.

Reply:
The ranges get sorted after loaded data is inserted into the loaded data vector.

Reply:
Okay, as long as you can maintain both orderings at the same time then it's fine.